search.js

```diff
@@ -1,18 +1,16 @@
 module.exports = function search (haystack) {
   haystack = haystack.toLowerCase()
   var self = {
-    for: (needle, fromIndex) => {
-      var foundStart = haystack.indexOf(needle[0].toLowerCase(), fromIndex || 0)
-      var notFound = foundStart === -1
-      var lastIteration = needle.length === 1
-      if (notFound) { return { found: false } }
-      if (lastIteration) {
-        return { found: true, start: foundStart, end: foundStart + 1 }
+    for: (needle) => {
+      var foundIndex = 0
+      var result = {found: true}
+      for (var i = 0; i< needle.length; i++){
+        foundIndex = haystack.indexOf(needle[i].toLowerCase(), foundIndex)
```

Comment:
you can move the `toLowerCase()` out of the loop since it's recursive, like we did we `haystack`.